src/js/jsx/search/SearchBar.jsx

```diff
@@ -68,8 +71,8 @@ define(function (require, exports, module) {
             }
 
             var idArray = id.split("_"),
-                type = idArray.length === 2 ? idArray[0] : "",
-                idInt = idArray.length === 2 ? parseInt(idArray[1]) : parseInt(id),
+                type = idArray.length > 0 ? idArray[0] : "",
+                idInt = idArray.length > 1 ? parseInt(idArray[1]) : parseInt(id),
```

Comment:
I know this doesn't fully address your concern, but (for now) I changed this so that all IDs are now "<type>_<number>" whereas before the recent documents would have a file path string instead of a number, so at least now it's a bit more consistent in format. The Select component requires a string ID, so the other ideas about fixing this would require a lot of changes in shared components.